mmv1/third_party/terraform/transport/config.go.tmpl

```diff
@@ -1156,6 +1224,26 @@ type StaticTokenSource struct {
       oauth2.TokenSource
 }
 
+// getExternalAccountConfig returns an externalaccount.Config based on the Config object.
+// The external account config is intended to be used to create a token source.
+func (c *Config) getExternalAccountConfig(clientScopes []string) externalaccount.Config {
+	eaConfig := externalaccount.Config{
```

Comment:
This is the part of the code with the biggest set of remaining unknowns - what pre-existing provider arguments should impact this externalaccount.Config? Currently we use the `clientScopes` argument, which is created using any values supplied via `scopes` in the provider block, but there are possibly other things to include:

1) Universe Domains

How should the [universe_domain provider argument](https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/provider_reference#universe_domain-1) be used here? How would we test that? There is current work in progress to add test coverage for use of universe_domain, so we would not be able to test use of `external_credentials_hcp_terraform` when `universe_domain` is also set.

Maybe we should purposefully not make this feature compatible with universe_domain at first, due to inability to test?

2) QuotaProjectID

I believe this would be dependent on user_project_override and billing_project.

3) ServiceAccountImpersonationLifetimeSeconds

Do we want to set this?